src/melee.cpp

```diff
@@ -539,9 +540,32 @@ bool Character::melee_attack_abstract( Creature &t, bool allow_special,
         add_msg( m_bad, _( "This weapon is too unwieldy to attack with!" ) );
         return false;
     }
-
     int move_cost = attack_speed( *cur_weapon );
 
+    if( is_avatar() && move_cost > 1000 && calendar::turn > melee_warning_turn ) {
+        const auto &action = query_popup()
+                             .preferred_keyboard_mode( keyboard_mode::keycode )
+                             .context( "CANCEL_ACTIVITY_OR_IGNORE_QUERY" )
+                             .message( pgettext( "cancel_activity_or_ignore_query", "<color_light_red>%s %s</color>" ),
+                                       string_format(
+                                           _( "Attacking with your %1$s will take a long time." ),
+                                           cur_weapon->display_name() ), "Are you sure you want to continue?" )
+                             .option( "YES", input_context::allow_all_keys )
+                             .option( "NO", input_context::allow_all_keys )
+                             .option( "IGNORE", input_context::allow_all_keys )
```

Comment:
```suggestion
                             .option( "YES" )
                             .option( "NO" )
                             .option( "IGNORE" )
```
No need for `input_context::allow_all_keys`.